packages/lib/reducer.ts

```diff
@@ -927,7 +928,9 @@ const reducer = produce((draft: Draft<State> = defaultState, action: any) => {
 			break;
 
 		case 'TAG_UPDATE_ALL':
-			draft.tags = action.items;
+			if (!isDeepStrictEqual(original(draft.tags), action.items)) {
+				draft.tags = action.items;
+			}
```

Comment:
It's strange that we'd need that because the point of immer is that it already performs this check for us. Otherwise we'd need to have everywhere `if (value !== previousValue) draf.value = value` but of course we don't (and if needed that, we may as well remove the immer dependency since it would no longer be useful).

Reply:
Not strange. Immer will find changes according to reference equality, but it doesn't check the deep equality of the found changes.

`if (value !== previousValue) draf.value = value` is not needed, because it checks reference equality which is performed by Immer.

Reply:
Right, that's correct, I was confused. This reducer is also used in the mobile app and I don't think React Native has this "utils" module. We have "fastDeepEqual" which I think you can use instead.

By the way, I guess it should be easy to unit test this? You just do `state = reducer(state, action);` and compete the state before and after.

Reply:
> This reducer is also used in the mobile app and I don't think React Native has this "utils" module.

Sorry, I wasn't aware of the point.

Finished. d1e7816